app/views/govuk_publishing_components/components/docs/layout_header.yml

```diff
@@ -28,6 +28,12 @@ examples:
     data:
       environment: production
       product_name: Product
+  with_product_name_on_public:
+    description: By default when the environment is 'public', the product name is not shown. This option overrides that.
+    data:
+      environment: public
+      product_name: Product
+      show_product_name: true
```

Comment:
That's a fair comment. My concern was that I not break anything. Initially I changed the logic by simply moving the `unless` line to wrap only the 'environment' bit, but that made it suddenly appear in the '[with search left and navigation](https://components.publishing.service.gov.uk/component-guide/layout_header/with_left_search_and_navigation)' option, and I was worried that would make it appear in places where it wasn't wanted.

My knowledge of the use of this component is pretty limited though, so if going simpler as described and not introducing a new option is better, I'm happy to amend.

Reply:
Sure, might be worth checking with @alex-ju as he introduced the change before. I can understand your concern about not breaking things but good to see if we can avoid making the API convoluted, that may mean having to make a breaking change, but that's probably better for future maintenance and iteration.

I can see this is used on the Coronavirus services, e.g: https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form/blob/092e527b41aa656a822154c442c9822976ea5de5/app/views/layouts/application.html.erb#L44 that would be affected.

I assume the initial change was made because the arguments don't cope with product_name missing and will insert "Publishing" as a name. This default ends up feeling increasingly weird as this component is used in non admin app contexts. I think only Content Publisher actually makes use of this default - and collections publisher actively tries to defeat it: https://github.com/alphagov/collections-publisher/blob/4895f0006085953180c63fcde09c3507cbee256d/app/views/layouts/admin_layout.html.erb#L12

Reply:
There's definitely some oddness to how this works. The thing that made me cautious in the first place was that when I first did this I changed it to simply show the product name if it was passed, but that made a product name appear in [this example](https://components.publishing.service.gov.uk/component-guide/layout_header/with_left_search_and_navigation) - even though it's not actually set, so it must be picking it up from somewhere else.

Additionally, In the `_layout_for_admin` component, the product name gets [set by default](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_layout_for_admin.html.erb#L1) to `Publishing`, which is exactly what the `_layout_header` component then [does when it's included inside the layout](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_layout_header.html.erb#L2).

I did talk to @alex-ju about this and he couldn't think of a reason not to simply show the product name if it gets set, but then I looked at the code and got worried I'd be breaking something.

This has been helpful. I'm going to change this PR. I'll simplify the logic so if a product name is passed, it gets shown.  This will mean some changes, but I'll check through the other examples you listed and make sure it's all sensible.

Reply:
@kevindew have modified and reworded the description of this PR to be more helpful. Do you think it's worth modifying the layout component for admin to not automatically set a product name?

Reply:
That's great. Yes I think applying the same approach to the admin layout makes sense too.

Would you be able to also make environment not be magical too?

As I'm something of the instigator of the breaking part of this I'm happy to do the work fixing repos that are affected by this.

Reply:
Okay, I _think_ I've understood the code and implemented the changes correctly, can you take a look please @kevindew?